mattermost/focalboard
# Mattermost auth integration

This allows Focalboard to authenticate via Mattermost, and provides a workspace per channel. To set up:
* Set `"authMode": "mattermost"` in config.json
* Set `"mattermostURL"` to your Mattermost test server's root url (e.g. "https://my.mattermost.com")
* Create an OAuth app in Mattermost, and populate `mattermostClientID` and `mattermostClientSecret`
* (If you have an existing install, clear your browser session, and delete the sessions table to avoid infinite loops - may have some to fix later)

To test:
* Get the channelID of a channel on your test server (from view channel info)
* Navigate to `http://localhost:8000/workspace/<channelID>`
* It should go through the OAuth loop and show an empty workspace
* Boards are scoped to each workspace

Breaking changes to the REST API:
* Add a `/workspaces/<workspaceID>/` prefix to the block APIs

We might need a few passes of reviews. To start, please review the overall login flow (in `mmauth.go`).
